### DISCUSSION TITLE
consult about bluetooth in esp32 of newbie in micropython

### DISCUSSION THREAD

**Discussion by jummiGz** [micropython/micropython#11729] [ESP32] [answered]
I'm using a Lolin S2 Mini board with the latest version of micropython, v1.20.
I want to learn how to use the bluetooth on this board, but I am not able to get any of the examples in the documentation to work.

All the examples ask to load either a bluetooth or ubluetooth library, but none of them are available in the version I'm using.

Could you clarify how I can use, in a simple way, the bluetooth of my device?

Some example?

**Comment by 2dof** [comment]
I haven't work with Lolin S2 Min but it is based on  esp32-s2  so You can try start from [Nordic UART Service ](https://github.com/2black0/MicroPython-ESP32-BLE) example at beginning.   
Since it is BLE then Serial Bluetooth terminal app for android is excellent for start. 
For windows/ mac os sometimes a dongle is needed (depends from laptop/pc type) and then using python (for example bleak lib)  more complicated examples can be mage.

**Comment by jummiGz** [comment]
Yes, the information that indicates I had already seen.
The steps that I do to be able to work with this example are the following:

1 .- I download the file main.py
2.- I save it in the esp32
3.- restart

The error that you give me is:
 Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
ImportError: no module named 'ubluetooth'

 In line 3 I see that it is :  import ubluetooth

And that's where I can't move forward, since I can't find where I can download those libraries.

**Reply by 2dof** [reply, marked as answer]
Try load another firmware version , for example [MicroPython v1.19.1](https://micropython.org/download/LOLIN_S2_MINI/).

edit:  I check manifest file for your board, and  propably ble is not  available  ( verison C3 has BLE but S2 not)

**Reply by jummiGz** [reply]
> Try load another firmware version , for example [MicroPython v1.19.1](https://micropython.org/download/LOLIN_S2_MINI/).
> 
> edit: I check manifest file for your board, and propably ble is not available ( verison C3 has BLE but S2 not)

I thought so, but since I'm a newbie I wasn't sure.

**Comment by jummiGz** [comment]
I have tried version 1.19 and 1.18 and neither of them works. The example, it seems, is for version 1.14, but for espS2 there is no version below 1.18, so I can't test it.

  Versions 1.18, 1.19 and 1.20 don't work with the example you gave me.

**Reply by 2dof** [reply]
I have done double check -> unfortunately there is a matter of board version,  micropython for s2 board version do not support bluetooth ( you can check [board.json](https://github.com/micropython/micropython/blob/master/ports/esp32/boards/LOLIN_S2_MINI/board.json) . Also in description in wendors , there is no info abaut suporting BT , only Wi-fi.

**Comment by jummiGz** [comment]
Thank you for taking the time to help me.
Thank you so much.

I hope they will be able to support this board soon.
All the best.
